main.tf

```diff
@@ -84,19 +90,45 @@ resource "aws_security_group" "emr_slave" {
 # EMR resources
 #
 resource "aws_emr_cluster" "cluster" {
-  name           = "${var.name}"
-  release_label  = "${var.release_label}"
-  applications   = "${var.applications}"
-  configurations = "${var.configurations}"
+  name                              = "${var.name}"
+  release_label                     = "${var.release_label}"
+  applications                      = "${var.applications}"
+  configurations                    = "${var.configurations}"
+  keep_job_flow_alive_when_no_steps = "${var.keep_job_flow_alive_when_no_steps}"
 
   ec2_attributes {
     key_name                          = "${var.key_name}"
     subnet_id                         = "${var.subnet_id}"
+    service_access_security_group     = "${var.service_access_security_group_id}"
+    additional_master_security_groups = "${var.service_access_security_group_id}"
```

Comment:
Appreciate you might only have one security group ID, but you might want to replicate it in idffernt variable name, so the var name matches the ec2_attribute name.